File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/tasks/tilecache/ExportTileCacheJobNativeObject.java
package com.valentingrigorean.arcgis_maps_flutter.tasks.tilecache;

import com.esri.arcgisruntime.tasks.tilecache.ExportTileCacheJob;
import com.valentingrigorean.arcgis_maps_flutter.concurrent.JobNativeHandler;
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.ArcgisNativeObjectsController;
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.BaseNativeObject;
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.NativeHandler;
import com.valentingrigorean.arcgis_maps_flutter.io.RemoteResourceNativeHandler;

public class ExportTileCacheJobNativeObject extends BaseNativeObject<ExportTileCacheJob> {

    public ExportTileCacheJobNativeObject(String objectId, ExportTileCacheJob job, ArcgisNativeObjectsController.NativeObjectControllerMessageSink messageSink) {
        super(objectId, job, new NativeHandler[]{
                new JobNativeHandler(job, JobNativeHandler.JobType.EXPORT_TILE_CACHE),
                new RemoteResourceNativeHandler(job),
        });
        setMessageSink(messageSink);
    }
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/tasks/tilecache/ExportTileCacheJobNativeObject.kt
package com.valentingrigorean.arcgis_maps_flutter.tasks.tilecache

import com.esri.arcgisruntime.tasks.tilecache.ExportTileCacheJob
import com.valentingrigorean.arcgis_maps_flutter.concurrent.JobNativeHandler
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.BaseNativeObject
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.NativeHandler
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.NativeMessageSink
import com.valentingrigorean.arcgis_maps_flutter.io.RemoteResourceNativeHandler

class ExportTileCacheJobNativeObject(
    objectId: String?,
    job: ExportTileCacheJob,
    messageSink: NativeMessageSink?
) : BaseNativeObject<ExportTileCacheJob?>(
    objectId, job, arrayOf<NativeHandler>(
        JobNativeHandler(job, JobNativeHandler.JobType.EXPORT_TILE_CACHE),
        RemoteResourceNativeHandler(job)
    )
) {
    init {
        setMessageSink(messageSink)
    }
}